crates/kubelet/src/provider/mod.rs

```diff
@@ -135,6 +135,38 @@ pub trait Provider: Sized {
     }
 }
 
+/// Resolve the environment variables for a container.
+///
+/// This generally should not be overwritten unless you need to handle
+/// environment variable resolution in a special way, such as allowing
+/// custom Downward API fields.
+///
+/// It is safe to call from within your own providers.
+pub async fn env_vars(
```

Comment:
note: this comes from the Provider trait, a few lines above. Now that the logic is internal to the state machine transitions rather than the provider, we should look into refactoring this out of `mod provider` so it can be accessible to all state machines.